### DISCUSSION TITLE
Is the goup.by function of analyse_sc_clusters() available now?

### DISCUSSION THREAD

**Discussion by bravedu01** [reactome/ReactomeGSA#52] [General]
The goup.by function of analyse_sc_clusters() will be highly helpful to compare the treatment with control group.

**Comment by jgriss** [comment]
Thanks for the feedback @bravedu01 !

As you probably now, this can currently be easily achieved by adapting the `Ident` (for Seurat objects). We are aware that this is an extra line of code but that's the reason why we haven't tackled this feature yet.

**Reply by bravedu01** [reply]
Thanks for the suggestion.

I know setting `ident` will be able to compare the treatment with control for the whole object. Is there a way to compare the treatment with control for all the subtypes. eg.  analyse_sc_clusters(pbmc, group.by="Group"), could we get the comparison of pathways between treatment and control in B/T/NK/Mono subtypes of pbmc?

The advantage of doing this is that we could obtain a heatmap comparing the selected pathway across treatments and subclusters(two layers at the same time) and figure out which subcluster is the most interested one.

**Comment by jgriss** [comment]
Hi @bravedu01 

Yes, you just have to adapt the primary ident:

```r
# save the current ident
seurat_obj$old_ident <- Idents(seurat_obj)

# set "Group" as primary ident
Idents(seurat_obj) <- seurat_obj$Group

# now run ReactomeGSA
```
